src/Pact/Native/Db.hs

```diff
@@ -216,15 +216,22 @@ descModule i [TLitString t] = do
     Nothing -> evalError' i $ "Module not found: " <> pretty t
 descModule i as = argsError i as
 
--- | unsafe function to create domain from TTable.
-userTable :: Show n => Term n -> Domain RowKey RowData
-userTable = UserTables . userTable'
+-- | Create domain from table
+userTable :: Term n -> Eval e (Domain RowKey RowData)
+userTable t = UserTables <$> userTable' t
 
--- | unsafe function to create TableName from TTable.
-userTable' :: Show n => Term n -> TableName
-userTable' TTable {..} = TableName $ asString _tModuleName <> "_" <> asString _tTableName
-userTable' t = error $ "creating user table from non-TTable: " ++ show t
+-- | Munge table name per environment.
+userTable' :: Term n -> Eval e TableName
+userTable' TTable {..} = view eeTableMunger >>= \f -> pure $ f _tModuleName _tTableName
+userTable' t = evalError' t "userTable invariant error"
```

Comment:
Note this was `error` before and would have only happened from an internal error so should not be in replay.